### PR TITLE
Give a more decent look to the GtkDialog.

### PR DESCRIPTION
**Pull request by TurboGit**
Dialogs for importing images or folders or XMP are now readable. Could
probably be better.

Another CSS work to make dt dialogs usable.

### CODE REVIEW

File: data/darktable.css.in [hunk 1]
```diff
@@ -84,15 +84,15 @@ GtkAlignment
   background-color: shade(@selected_bg_color, 1.2);
   background-image: none;
 }
-#lib-plugin-ui .button:${GTK_CSS_CLASS},
-.button:${GTK_CSS_CLASS}
```
Comment (LebedevRI) [inline data/darktable.css.in]: Why do you think there was ${GTK_CSS_CLASS} ?

Comment (TurboGit) [inline data/darktable.css.in]: I know, to accommodate different Gtk+ version. I didn't push the right version. Will fix that in a moment.

**Comment by TurboGit** [comment]
Should be ok now.

**Comment by LebedevRI** [comment]
To be rebased.

IMHO, while proposed version looks a bit better *in some parts*, in other it looks even worse.
import-master
import-pr816

**Comment by TurboGit** [comment]
Another attempt at this. This time I have kept the native GtkDialog which looks nice after all.

**Comment by TurboGit** [comment]
Rebased on master. I'll merge very soon as it is painful to handle merge conflicts.

**Comment by TurboGit** [comment]
Now merged. I'm sure we can still improve all this, but it is getting better.

**Comment by LebedevRI** [comment]
Regression: tooltips now look different.

**Comment by TurboGit** [comment]
I'll have a look.

2015-01-08 14:27 GMT+01:00 Roman Lebedev <notifications@github.com>:

> Regression: tooltips now look different.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/darktable-org/darktable/pull/816#issuecomment-69178518>
> .
>



-- 
  Pascal Obry /  Magny Les Hameaux (78)

  The best way to travel is by means of imagination

  http://v2p.fr.eu.org
  http://www.obry.net

  gpg --keyserver keys.gnupg.net --recv-key F949BD3B

**Comment by LebedevRI** [comment]
@TurboGit lib/iop preset menu is severely regressed too.
If this is not fixed soon, i'm reverting.

**Comment by TurboGit** [comment]
Which version of Gtk?

To be sure we are seeing the same thing. Here is a copy of my tool-tips. Looks nice no? Do you have the same on your side?

The preset menu is the same color for the background/foreground. Cannot take a snapshot while it is opened.
capture d ecran de 2015-01-08 15 20 25

**Comment by LebedevRI** [comment]
@TurboGit
I have said *preset*. That is completely unreadable to me:
preset-regression

**Comment by TurboGit** [comment]
Which Gtk version?

I don't have this display at all. As I said I have the same display for preset than for tool-tip.

I'll find a way to fix that but please do not revert! We are making progress this is just a "small" side effect if you compare the unusable dialogs we had before.

**Comment by TurboGit** [comment]
@BTW, you're sure you don't have any local CSS in ~/.config/darktable/darktable.css?

**Comment by TurboGit** [comment]
For me in fact I'm now following the color as set for the theme at the top of our CSS file:

@define-color tooltip_bg_color #000000;
@define-color tooltip_fg_color #ffffff;

And this has not changed since the first version.

**Comment by LebedevRI** [comment]
Debian sid, gtk+-3.14.5
```
$ ls -lah ~/.config/darktable/darktable.css
ls: cannot access /home/lebedevri/.config/darktable/darktable.css: No such file or directory
```
If i change gtk theme, that unreadable text color/bgcolor changes, so you are clearly not overriding enough colors from theme defaults.
If i revert 6233506d62f12441a528881cdda9c7649bb7710e, that preset menu always looks the same, no matter the gtk+ theme, so it is a regression.

**Comment by TurboGit** [comment]
Debian sid, gtk+-3.14.5 ... same version here. I'll check how to override the tool-tip CSS.

**Comment by TurboGit** [comment]
should be fixed. let me know.

**Comment by LebedevRI** [comment]
Preset menu seems to look as before 6233506d62f12441a528881cdda9c7649bb7710e.
Not sure we like that tooltips now have a solid white double outline.

**Comment by TurboGit** [comment]
Good to hear!

For the tooltip I have introduced a rule to make it easy to change them. I'm not sseing the white double outline, so I suppose that's again some missing CSS directives in .tooltip.

**Comment by LebedevRI** [comment]
@TurboGit settings look wrong too.
I strongly suggest you to look at *ALL* widgets and make them look as they were.

**Comment by TurboGit** [comment]
I see, settings is a GtkDialog so now it is using the native look. Is that not acceptable? After all it does not look like before but I find the layout quite nice. If not it should be possible to give it a flat look as it has a special name, but this will make the CSS heavier....

**Comment by hanatos** [comment]
2015-01-09-060705_1920x1080_scrot

this is what it looks for me. i think it looks quite unacceptable with all the different bg colours.

also i really don't like how the buttons don't look flat.

**Comment by TurboGit** [comment]
ok, will do the necessary changes to get back to a flat look.

**Comment by TurboGit** [comment]
How does it looks to you know?

**Comment by hanatos** [comment]
background in prefs looks better, i still think the whole idea to leave
anything to the system style is a bad idea. all colours look different and
different button styles look like it doesn't belong to the same application
any more. the one thing it seems to fix is the padding of all gui elements
in the gtkfilechooser. this could probably have been achieved in much less
intrusive ways, without the side effect of breaking look and feel for
gtkdialogs.

**Comment by hanatos** [comment]
2015-01-09-091910_1920x1080_scrot

i noticed some #preferences_notebook .button in the css.. are these intended to also affect the close button at the bottom? because it doesn't here..

**Comment by TurboGit** [comment]
The button should be flat now. I agree that it would have probably be better to not specialized the GtkDialog, I tried (hard) but felt to have something usable. I may try again at some point...

**Comment by hanatos** [comment]
looks great thanks. yeah, css should support regular expressions and
negation, not just this subset that it does.. :)

**Comment by LebedevRI** [comment]
@TurboGit That didn't fix all the buttons, e.g. the ones in lens iop are still not flat.

**Comment by hanatos** [comment]
also another success story of a different system with different default css in the background. i very much doubt leaving anything to the default system is a good idea.

2015-01-09-152615_2560x1440_scrot

**Comment by TurboGit** [comment]
Ok, at this point let's face it... I failed :( I'll revert.

**Comment by TurboGit** [comment]
Reverted.

**Pull request closed without merging**